core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -162,6 +162,7 @@ class Partition(val topic: String,
    */
   def makeLeader(controllerId: Int,
                  partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
+    var leaderHWIncremented = false
```

Comment:
I'm totally unconvinced by the argument. :) Returning a value cannot be more obscure than mutating a variable in an outer scope as it makes it easier to reason about the code (particularly from a code evolution/refactoring perspective).

It goes without saying that I'm a big fan of value-based programming and I suspect you'd change your mind on the idiomatic Scala code point given enough exposure. :)